crypto/evp/pkey_mac.c

```diff
@@ -250,10 +248,12 @@ static int pkey_mac_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
             hctx->raw_data.md = p2;
             break;
         case MAC_TYPE_MAC:
-            if (ctx->pkey != NULL
-                && !EVP_MAC_CTX_copy(hctx->ctx,
-                                     (EVP_MAC_CTX *)ctx->pkey->pkey.ptr))
-                return 0;
+            if (ctx->pkey != NULL) {
+                /* Can leak memory? */
```

Comment:
Yup. That's what happens with the *one* CMAC test that has Travis fail. You must free `hctx->ctx` before assigning the new duplicate.

Reply:
It's it great that can add a comment at the exact place where you're leaking. I didn't fully understand part of the code at that point in time.